Update sort-keys-recursive to version 2.0.0

matteocng opened the pull request:
Commits:
- docs: fix typos and rearrange some sentences
- feat: update sort-keys-recursive to version 2.0.0

Fixes #35.

matteocng committed:
docs: fix typos and rearrange some sentences

Kikobeats commented on lib/Finepack.coffee:
```diff
@@ -45,10 +46,10 @@ module.exports = (data, opts = {}, cb) ->
+  input = sort input, opts.sortOptions
 
+  # After sorting, we move some fields in new positions for esthetic effect.
   for key in KEYWORDS.sort when input[key]?
```
Do you think we can avoid this piece of code? I mean, same behavior, but using `sort-keys-recursive` benefits.

matteocng replied:
Do you mean refactoring the code that sorts by `KEYWORDS.sort` or just removing that comment?

Kikobeats replied:
the first, but not necessary, now works.

Kikobeats commented:
Thanks for that. I added comments 😄

Kikobeats commented:
Can I wait for something more or time to merge? 😄

matteocng commented:
Sorting a second time using `sort-keys-recursive` and passing `compareFunction` works but it doesn't seem ideal, because the function would be applied to all keys recursively, not just the keys at the first level.

What do you think about this instead:
```coffeescript
# After sorting, we move some fields in new positions for esthetic effect.
compareFunction = (a, b) -> KEYWORDS.sort.indexOf(a) - KEYWORDS.sort.indexOf(b)
output = Object.keys(input).sort(compareFunction).reduce (acc, key) ->
 acc[key] = input[key]
 acc
, {} # Object, second argument of reduce(), initial value of 'acc'.
```

Kikobeats commented:
@matteocng  I like this new approach, can you commit it? want to know if tests are green with this change

matteocng committed:
feat: update sort-keys-recursive to version 2.0.0

- Add 'sortOptions' (Object) key in 'options', and pass it to sort-keys-recursive.
- Add docs and tests for all available 'sortOptions'.
- Warn about the behavior of JavaScript's 'sort' function in the comments and 'README.md'.
- Add '--sort-ignore-object-at' and '--sort-ignore-array-at' command-line options.
- Replace: 'var' with 'const'/'let', 'function' with arrow function, {foo: foo} with {foo}.
- Minor modifications to comments.

matteocng commented:
I have updated the commit, also adding a small clarification about the behaviour of the default sort function (and link to MDN) in the `README` and code comments.
